Library/Homebrew/attrable.rb

```diff
@@ -0,0 +1,32 @@
+# typed: strict
+# frozen_string_literal: true
+
+module Attrable
+  sig { params(attrs: Symbol).void }
+  def attr_predicate(*attrs)
+    attrs.each do |attr|
+      define_method attr do
+        instance_variable_get("@#{attr.to_s.sub(/\?$/, "")}") == true
+      end
+    end
+  end
+
+  sig { params(attrs: Symbol).void }
+  def attr_rw(*attrs)
+    attrs.each do |attr|
+      module_eval <<-EOS, __FILE__, __LINE__+1
```

Comment:
Can this be done with `define_method`?

Reply:
Done! (I simplified it as well, taking advantage of `instance_variable_get` returning `nil` for unset ivars.)